src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -216,25 +216,36 @@ public CopyPosition CopyTo(CopyPosition position, T[] array, int arrayIndex, int
             int row = position.Row;
             int column = position.Column;
 
-            for (; count > 0; row++, column = 0)
+            T[] buffer = GetBuffer(row);
+            int copied = CopyToCore(buffer, column);
+
+            while (count > 0)
+            {
+                buffer = GetBuffer(++row);
+                copied = CopyToCore(buffer, 0);
+            }
```

Comment:
We don't have any tests that enter this loop.

Reply:
Added tests with larger data sets